From 60f86a91bf361342bddcfbe5804f00eb2c8e7eb7 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Fri, 2 Feb 2018 05:19:09 -0800
Subject: [PATCH] Attempt to fix issues/530

* DONE (IRkernel)
Making 'packages.html' ...Error in flush.console() : attempt to apply non-function
Making 'packages.html' ...Error in flush.console() : attempt to apply non-function
Execution halted
---
 R/environment_shadow.r | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/R/environment_shadow.r b/R/environment_shadow.r
index 34f645d..e275f45 100644
--- a/R/environment_shadow.r
+++ b/R/environment_shadow.r
@@ -73,6 +73,9 @@ init_shadowenv <- function() {
         stop(sQuote('edit()'), ' not yet supported in the Jupyter R kernel')
     })
     
+    # https://github.com/IRkernel/IRkernel/issues/530#issuecomment-350997898
+    if (is.null(backup_env$utils_flush_console)) backup_env$utils_flush_console <- identity
+
     # stream output in loops:
     # https://github.com/IRkernel/IRkernel/issues/3
     replace_in_package('base', 'flush.connection', function(con) {
-- 
2.15.0

